Syntaxes/References.sublime-syntax

```diff
@@ -1,34 +1,47 @@
 %YAML 1.2
 ---
+# [Subl]: https://www.sublimetext.com/docs/3/syntax.html
+# [LSP]: https://github.com/Microsoft/language-server-protocol/blob/master/protocol.md
 hidden: true
 scope: output.lsp.references
 
+variables:
+  start_of_reference_body: ^\s+(?=\d)
+  filename_and_colon: ^\s*(\S)\s+(.*)(:)$
+
 contexts:
   main:
     - include: references-preamble
     - include: references-body
 
   references-preamble:
-    # References to "function" at path\file.py:
-    - match: ^(.*)(:)$
+    - match: '{{filename_and_colon}}'
       captures:
-        0: meta.references.preamble.lsp
-        1: string.unquoted.lsp
-        2: punctuation.separator.lsp
+        0: meta.reference.preamble.lsp
+        1: punctuation.section.references.preample.lsp
+        2: string.unquoted.lsp
```

Comment:
I would even go as far and assign `entity.name.file.references.lsp` here.

Reply:
Is this alright? 
```diff
    0: meta.reference.preamble.lsp
    1: punctuation.section.references.preample.lsp
-   2: string.unquoted.lsp
+   2: string.unquoted.lsp entity.name.file.references.lsp 
    3: punctuation.separator.lsp
```

Reply:
Yeah, that's good. This way it even has a fallback scope with the string.